frame/support/procedural/src/pallet/parse/storage.rs

```diff
@@ -75,12 +77,45 @@ impl syn::parse::Parse for PalletStorageAttr {
 			})?;
 
 			Ok(Self::StorageName(renamed_prefix, attr_span))
+		} else if lookahead.peek(keyword::unbounded) {
+			content.parse::<keyword::unbounded>()?;
+
+			Ok(Self::Unbounded(attr_span))
 		} else {
 			Err(lookahead.error())
 		}
 	}
 }
 
+struct PalletStorageAttrInfo {
+	getter: Option<syn::Ident>,
+	rename_as: Option<syn::LitStr>,
+	unbounded: bool,
+}
+
+impl PalletStorageAttrInfo {
+	fn from_attrs(attrs: Vec<PalletStorageAttr>) -> syn::Result<Self> {
+		let mut getter = None;
+		let mut rename_as = None;
+		let mut unbounded = None;
+		for attr in attrs {
+			match attr {
+				PalletStorageAttr::Getter(ident, ..) if getter.is_none() => getter = Some(ident),
+				PalletStorageAttr::StorageName(name, ..) if rename_as.is_none() =>
+					rename_as = Some(name),
+				PalletStorageAttr::Unbounded(..) if unbounded.is_none() => unbounded = Some(()),
+				attr =>
+					return Err(syn::Error::new(
+						attr.attr_span(),
+						"Invalid attribute: Duplicate attribute",
+					)),
+			}
+		}
+
+		Ok(PalletStorageAttrInfo { getter, rename_as, unbounded: unbounded.is_some() })
```

Comment:
```suggestion
		let mut unbounded = false;
		for attr in attrs {
			match attr {
				PalletStorageAttr::Getter(ident, ..) if getter.is_none() => getter = Some(ident),
				PalletStorageAttr::StorageName(name, ..) if rename_as.is_none() =>
					rename_as = Some(name),
				PalletStorageAttr::Unbounded(..) if !unbounded => unbounded = true,
				attr =>
					return Err(syn::Error::new(
						attr.attr_span(),
						"Invalid attribute: Duplicate attribute",
					)),
			}
		}

		Ok(PalletStorageAttrInfo { getter, rename_as, unbounded: unbounded })
```

why are masking a `bool` with `Option<()>` here?

Reply:
yes, because i was thinking it is about the existence of the attribute, but bool is good indeed.